vendor/github.com/OpenBazaar/spvwallet/peers.go

```diff
@@ -236,124 +209,75 @@ func (pm *PeerManager) onVerack(p *peer.Peer, msg *wire.MsgVerAck) {
 		p.NA().HasService(SFNodeBitcoinCash) { // Don't connect to bitcoin cash nodes
 		// onDisconnection will be called
 		// which will remove the peer from openPeers
+		log.Warningf("Peer %s does not support bloom filtering, diconnecting", p)
 		p.Disconnect()
 		return
 	}
 	log.Debugf("Connected to %s - %s\n", p.Addr(), p.UserAgent())
 	// Tell the addr manager this is a good address
 	pm.addrManager.Good(p.NA())
-
-	filter, err := pm.getFilter()
-	if err != nil {
-		log.Error(err)
-		return
+	if pm.msgChan != nil {
+		pm.msgChan <- newPeerMsg{p}
 	}
-	p.QueueMessage(filter.MsgFilterLoad(), nil)
-
-	pm.peerMutex.Lock()
-	pm.readyPeers[p] = struct{}{}
-	if pm.downloadPeer == nil {
-		pm.setDownloadPeer(p)
-	}
-	pm.peerMutex.Unlock()
 }
 
 func (pm *PeerManager) onDisconnection(req *connmgr.ConnReq) {
 	// Remove from connected peers
 	pm.peerMutex.Lock()
 	defer pm.peerMutex.Unlock()
-	peer, ok := pm.openPeers[req.ID()]
+	peer, ok := pm.connectedPeers[req.ID()]
 	if !ok {
 		return
 	}
-
-	log.Debugf("Peer%d disconnected", peer.ID())
-	delete(pm.openPeers, req.ID())
-	delete(pm.downloadQueues, peer.ID())
-	delete(pm.readyPeers, peer)
-
-	// If this was our download peer we lost, replace him
-	if pm.downloadPeer != nil && peer != nil {
-		if pm.downloadPeer.ID() == peer.ID() {
-			close(pm.blockQueue)
-			go pm.selectNewDownloadPeer()
-		}
-	}
-}
-
-func (pm *PeerManager) selectNewDownloadPeer() {
-	for peer := range pm.readyPeers {
-		pm.setDownloadPeer(peer)
-		break
+	log.Debugf("Peer %s disconnected", peer)
+	delete(pm.connectedPeers, req.ID())
+	if pm.msgChan != nil {
```

Comment:
How can we get the `msgChan` nil check removed from every message? Any way we ensure that the `PeerManager` never has a nil `msgChan`? If not, then this should be in a private method that handles this for us quietly. Also, what does it mean for the msgChan to be nil and then drop the command it intended to send entirely...is this case handled?